labs/b4.md

```diff
@@ -0,0 +1,269 @@
+lab.
+
+## But what about software that isn’t in the repositories?
+
+Sometimes it happens that a program you want to install hasn't been
+packaged for your distributions, or hasn't been packaged at all. You have
+several options in this scenario to install the software you want:
+
+#### Warning:
+**Installing software not from a repository carries the same risks as
+installing software from a random .exe or .msi on Windows.**
+
+Linux is popularly considered more secure than Windows or MacOS because of
+its use of repositories -- but a malicious package installed manually can
+pwn your system as easily as malware on any other OSes.
+**Use common sense! Only manually install from sources you trust.**
+
+*(For users of Ubuntu: PPAs, or third-party repositories, carry the same
+risks of running malicious third-party code.)*
+
+If the developers provide a package compatible with your distribution (.deb for
+Debian-based distros), you can download that package and install it using:
+
+    $ sudo apt install ./package.deb
+
+Another way is to find a generic binary package from the developer. This can
+come in the form of shady shell scripts, or a binary tarball that you can just
+extract it and run, or an appimage, which is a special type of executable that
+includes its own dependencies.
+
+The last way is to compile your software from scratch. What does that mean?
+Open source software must have its source code publicly available somehow,
+(GitHub, GitLab, their website). If you fetch their source code, it
+won’t magically run out of the box. The source code is like the recipe, while
+the software itself is like food. A package is like a food that is put into a
+box but we won’t covering the details of making a package yourself (there are
+tools that do that for you and it varies from platform to platform).
+
+## So how do I compile?
+
+Compiling software on Linux can be a mixed bag. Sometimes, all the dependencies
+(like libraries) are installed on your computer and there is no fiddling around.
+Sometimes, the dependencies don’t even exist pre-compiled for your distribution
+so you have to compile those yourself in order to compile what’s at hand. Most
+of the time, these steps are simplified through the use of a Makefile, which
+controls the Make build system.
+
+In most source tarballs, there is usually a Makefile that contains a set of
+directives to compile a project. This is because there are usually multiple
+files across many directories that need to be compiled together into the
+final executable. On top of that, there are multiple settings that control, for
+example, optimizations, size of the executable, static vs. dynamic linking,
+and whether to link against system libraries or alternatives.
+
+Many projects that have to be compiled are usually in C, C++, or similar
+lower-level languages. On most Linux distributions, there are usually three
+compiler “options”. There is the GNU Toolchain which provides `gcc` or the GNU
```

Comment:
Either this paragraph should be up near `apt install gcc` or that command should be down here.